### PR TITLE
Do not use open coded `__host__ __device__` in tests

### PR DESCRIPTION
**Pull request by miscco**
We might want to use different attributes when tile comes around

### CODE REVIEW

**Review by davebayer — APPROVED** [review]
LGTM, but I'd like more opinions on this

File: libcudacxx/test/libcudacxx/cuda/annotated_ptr/access_property.pass.cpp [hunk 1]
```diff
@@ -11,7 +11,7 @@
 #include "utils.h"
 
 template <typename T>
-__host__ __device__ __noinline__ void test_global_implicit_property(T ap, cudaAccessProperty cp)
+_CCCL_API __noinline__ void test_global_implicit_property(T ap, cudaAccessProperty cp)
```
Comment (jrhemstad) [inline libcudacxx/test/libcudacxx/cuda/annotated_ptr/access_property.pass.cpp]: question: `_CCCL_API` doesn't feel very descriptive. Is there another macro name we could use that is more self-describing?

Comment (miscco) [reply libcudacxx/test/libcudacxx/cuda/annotated_ptr/access_property.pass.cpp]: Its the one we use internally. 

The issue is that we would otherwise make sure every file includes the "test_macros.h" file

Comment (fbusato) [reply libcudacxx/test/libcudacxx/cuda/annotated_ptr/access_property.pass.cpp]: I would prefer an alternative solution if possible
`_CCCL_API` has additional attributes. My concern is that this could even change the behavior in edge cases.
```cpp
#  define _CCCL_API        _CCCL_HOST_DEVICE _CCCL_VISIBILITY_HIDDEN _CCCL_EXCLUDE_FROM_EXPLICIT_INSTANTIATION
```

Comment (wmaxey) [reply libcudacxx/test/libcudacxx/cuda/annotated_ptr/access_property.pass.cpp]: I'd prefer if our tests implement the macros separately from CCCL. This provides some coverage in case there are differences.

Comment (miscco) [reply libcudacxx/test/libcudacxx/cuda/annotated_ptr/access_property.pass.cpp]: yeah it might just be a tad more work because not all tests include `test_macros.h` but we can do that

Comment (miscco) [reply libcudacxx/test/libcudacxx/cuda/annotated_ptr/access_property.pass.cpp]: Moved to `TEST_API` and `TEST_DEVICE_API`

File: libcudacxx/test/support/test_macros.h [hunk 2]
```diff
@@ -24,6 +24,10 @@
 #define TEST_DIAG_SUPPRESS_MSVC(...)  _CCCL_DIAG_SUPPRESS_MSVC(__VA_ARGS__)
 #define TEST_NV_DIAG_SUPPRESS(...)    _CCCL_BEGIN_NV_DIAG_SUPPRESS(__VA_ARGS__)
 
+// Use the CCCL host device function
+#define TEST_API        _CCCL_HOST_DEVICE
+#define TEST_DEVICE_API _CCCL_DEVICE
```
Comment (bernhardmgruber) [inline libcudacxx/test/support/test_macros.h]: Suggestion: I would prefer to have those macros named either `TEST_HOST_DEVICE` etc., or something that does not involve `API`, like `TEST_FUNC`.

Are we planning to add any other attributes except execution space annotations? Then I think I would prefer `TEST_[HOST_]DEVICE`, or just `TEST_HD`.

**Review by bernhardmgruber — APPROVED** [review]

File: libcudacxx/test/support/test_macros.h (hunk 2 above)
Comment (miscco) [reply libcudacxx/test/support/test_macros.h]: It will also container Tile, so I went with TEST_FUNC

**Review by davebayer — APPROVED** [review]
